media-proxy/src/mtl.c

```diff
@@ -1054,10 +1054,7 @@ int rx_shm_deinit(rx_session_context_t* rx_ctx)
         return -1;
     }
 
-    err = pthread_cancel(rx_ctx->memif_event_thread);
-    if (err) {
-        ERROR("%s: Error canceling thread: %s", __func__, strerror(err));
-    }
+    pthread_cancel(rx_ctx->memif_event_thread);
 
     err = pthread_join(rx_ctx->memif_event_thread, NULL);
```

Comment:
This could be even simplified like:

```c
// INFO: pthread_cancel returns only one error code: ESRCH
if (pthread_cancel(rx_ctx->memif_event_thread) != ESRCH) {
    err = pthread_join(rx_ctx->memif_event_thread, NULL);
}
if (err && err != ESRCH) {
    ERROR("%s: Error joining thread: %s", __func__, strerror(err));
}
```

Reply:
ok, I followed the suggestion

Reply:
at least the first one, it seems cleaner and more readable.